index.js

```diff
@@ -55,20 +55,29 @@ module.exports = function(search, replacement, options) {
         callback(null, file);
       }
 
-      if (options && options.skipBinary) {
-        istextorbinary.isText(file.path, file.contents, function(err, result) {
-          if (err) {
-            return callback(err, file);
+      if (options) {
+        if (options.passFilename && typeof replacement === 'function') {
+          var userReplacement = replacement;
+          replacement = function (match) {
+            userReplacement(match, file.path);
```

Comment:
Another potential solution would be to call the replacement function in the context of the file object:

```js
userReplacement.apply(file, arguments);
// this.path == the path of the file
```

Thoughts?